File: Maxglass.Ecommerce/Maxglass.Ecommerce.DataTransfer/Pedidos/Responses/PedidoResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Maxglass.Ecommerce.DataTransfer.Fretes.Responses;

namespace Maxglass.Ecommerce.DataTransfer.Pedidos.Responses
{
    public class PedidoResponse
    {
        public int Id { get; set; }
        public string? Cep { get; set; }
        public decimal? Valor { get; set; }
        public int? Situacao { get; set; }
        public string? NumeroEndereco { get; set; }
        public string? ComplementoEndereco { get; set; }
        public DateTime DataPedido { get;  set; }
        public DateTime DataPrevistaEntrega { get;  set; }
        public DateTime DataPrazo { get;  set; }
        public FreteResponse? Frete { get; set; }
        public IList<ItemPedidoResponse>? ItensPedido { get; set; }

        //public ClienteResponse? Cliente { get; set; }
    }
}

File: Maxglass.Ecommerce.Web/src/app/clientes/models/pedido.response.ts
import { SituacaoEnum } from './../enumeradores/situacao.enum';
export class PedidoResponse {
  id: number;
  cep: string;
  valor: number;
  situacao: SituacaoEnum;
  numeroEndereco: string;
  complementoEndereco: string;
  dataPedido: Date;
  dataPrevistaEntrega: Date;
  dataPrazo: Date;
  frete: any;
  itensPedido: Array<any>;
  isExpanded : boolean;

  constructor(params: Partial<PedidoResponse>) {
      this.id = params.id || 0;
      this.cep = params.cep || null;
      this.valor = params.valor || 0;
      this.situacao = params.situacao || 0;
      this.numeroEndereco = params.numeroEndereco || null;
      this.complementoEndereco = params.complementoEndereco || null;
      this.dataPedido = params.dataPedido || null;
      this.dataPrevistaEntrega = params.dataPrevistaEntrega || null;
      this.dataPrazo = params.dataPrazo || null;
      this.frete = params.frete || null;
      this.itensPedido = params.itensPedido || [];
      this.isExpanded = false;
  }
}

